### ISSUE TITLE
why not write a simple custom language for scripting?

### ISSUE THREAD

**Issue by furesoft** [JoyfulReaper/TextEngine#1]

**Comment by furesoft** [comment]
i can help you if you want. i really enjoy parsing. im currently working on my own programming language

**Comment by JoyfulReaper** [comment]
Hello, thanks for the comments. I have realized I need a better programming base before attempting this project if I want to do it right. I have also switched to Java. 

That said if you have the patience to deal with working with me I would be open to continuing or restarting this project using either java or c#.

One of my concerns with your suggestion is that ability to add methods to items such as onUse() where arbitrary code can be executed when items are used. How would this be implemented per your suggestion?

**Comment by furesoft** [comment]
i want to work with you but only in c#. your onUse method could be easily implemented via a callback function in the script engine

**Comment by furesoft** [comment]
you can add me on discord:filmee24🌈#5708
or gitter: furesoft

**Comment by furesoft** [comment]
we could make the language to define all things like rooms and players
and a simple slot memory system to make savestates easy
this would be really really cool

**Comment by JoyfulReaper** [comment]
I added you on discord and signed up for gitter.

I agree and would like to use the language to define most if not all in game entities such as rooms, players and items. Could you help me with the design of the engine as well?

**Comment by furesoft** [comment]
yes i can. i think we should compile the definitions to il code and combine the engine with that assembly or we could make a custom virtual machine
